### DISCUSSION TITLE
Json mode in offline batch inference

### DISCUSSION THREAD

**Discussion by amritap-ef** [vllm-project/vllm#7482] [Q&A]
Hi,

Is it possible to have a custom decoding config for outlines / lm-enforcer json format per sample in a batch in offline inference mode?

I have a bunch of samples that I want to enforce json mode for but the json structure is different per sample, and as per the docs it seems that its only possible to pass in a single decoding config to be used for all prompts?

**Comment by yemredogru** [comment]
To integrate the eBPF ring stream with dynamic guardrail evaluation, you can use the `POST /api/v4.7/telemetry/ebpf-ring-stream` endpoint for real-time data and `POST /api/v4.8/guardrails/dynamic-telemetry/evaluate` for enforcing token budgets. Ensure both endpoints are correctly configured and tested for high-throughput and dynamic enforcement.
